pkg/network/node/egressip.go

```diff
@@ -245,6 +245,10 @@ func (eip *egressIPWatcher) deleteNamespaceEgress(vnid uint32) {
 }
 
 func (eip *egressIPWatcher) claimEgressIP(egressIP, egressHex string) error {
+	if egressIP == eip.localIP {
```

Comment:
That means if the master is not also configured as a node, the master IP will still be able to set as egressIPs?

Reply:
The code for claiming egress IPs is part of the node code; if the master isn't a node then it can't claim egress IPs at all.

Reply:
Sorry, I meant that as the bug described. I can add the master host IP as the egressIPs to node which will make the node to master connection broken.

The fix is going to check the node hosts IPs should not be added as egressIPs. 
But for condition that the master only roles as master, but not node. Then the master host IP will still be able to add to nodes as egressIPs.

My env like:
host1: master
host2: node1
host3: node2

After the fix, the host1's IP will still be able to add to node1 or node2 since the host1 is not a node. But it will still break the node to master connection after that.

Reply:
Currently, we only validate EgressIP has valid IP (also IPv4 with https://github.com/openshift/origin/pull/16807) but I think we should ensure it does not overlap with other cidrs (node/service/ingress/external...) and master IPs.

Reply:
@bmeng: we can't really validate that the IP address is totally unused anyway; there might be other machines on the same subnet that aren't nodes/master. egress-router has the same issue. But in both cases, only cluster admins can set those IPs anyway, so we can just tell them "be careful"

Reply:
@pravisankar the EgressIP has to be on the same subnet as the node's primary IP, which we have already verified doesn't overlap with the other CIDRs.

Reply:
@danwinship Where did we check 'EgressIP is on the same subnet as the node's primary IP'?

Reply:
ah, indeed, we don't; we add it to the primary interface, so it probably won't *work* right otherwise, but we don't actually check it. I'll fix that